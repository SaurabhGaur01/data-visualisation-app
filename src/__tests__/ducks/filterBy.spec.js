import {
    setFilterBy,
    filterByReducer,
} from '../../ducks/filterBy';

describe('filterBy Duck Tests', () => {
    const mockKey = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setFilterBy should match the snapshot' , () => {
            expect(setFilterBy(mockKey)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(filterByReducer()).toEqual('construction');
        });
        it('should return the passed state if called with no action' , () => {
            expect(filterByReducer(mockKey)).toEqual(mockKey);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(filterByReducer(mockKey, unknownAction)).toEqual(mockKey);
        });
        it('should return the value which is passed' , () => {
            expect(filterByReducer('', setFilterBy(mockKey))).toEqual(mockKey);
        });
    });
});