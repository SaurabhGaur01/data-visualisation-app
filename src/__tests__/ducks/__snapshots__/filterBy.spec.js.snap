// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterBy Duck Tests Action Generator setFilterBy should match the snapshot 1`] = `
Object {
  "type": "modules/data-visualisation-app/SET_FILTER_BY",
  "value": "mockValue",
}
`;
